contracts/rewards/StableTriMaker.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: MIT
+// P1 - P3: OK
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../amm/interfaces/IUniswapV2ERC20.sol";
+import "../amm/interfaces/IUniswapV2Factory.sol";
+import "../amm/interfaces/IUniswapV2Router02.sol";
+import "../interfaces/ISwap.sol";
+
+// StableTriMaker is MasterChef's left hand and kinda a wizard. He can cook up Tri from pretty much anything!
+// This contract handles "serving up" rewards for xTri holders by trading stableswap tokens collected from fees for Tri.
+
+contract StableTriMaker is Ownable {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    IUniswapV2Router02 public immutable router;
+    address public immutable bar;
+    address private immutable tri;
+
+    event LogRemoveLiquidity(
+        address swapLpToken,
+        address[] stableTokensToRemoveTo,
+        uint256[] stableTokensToRemoveAmounts
+    );
+
+    event LogSwapStableTokenToTri(
+        address stableTokenToConvertToTri,
+        address[] triConversionPath,
+        uint256 stableTokenAmount
+    );
+
+    event LogTriSentToBar(uint256 triAmount);
+
+    constructor(
+        address _router,
+        address _bar,
+        address _tri
+    ) public {
+        router = IUniswapV2Router02(_router);
+        bar = _bar;
+        tri = _tri;
+    }
+
+    // C6: It's not a fool proof solution, but it prevents flash loans, so here it's ok to use tx.origin
+    modifier onlyEOA() {
+        // Try to make flash-loan exploit harder to do by only allowing externally owned addresses.
+        require(msg.sender == tx.origin, "StableTriMaker: must use EOA");
+        _;
+    }
+
+    function removeLiquidityFromFees(
+        address swap, // Stableswap Pool
+        address swapLpToken, // Stableswap LP token that this contract holds a non-zero balanceOf
+        address[] calldata stableTokensToRemoveTo // Stable tokens to remove from the LP at the same index
+    ) public onlyEOA returns (uint256[] memory) {
+        // Withdraw admin fees from the Stableswap Pool
+        ISwap(swap).withdrawAdminFees();
+        // Remove liquidity from stableswap lp into stableTokensToRemoveTo arg
+        uint256 stableTokensBalance = IERC20(swapLpToken).balanceOf(address(this));
+        IERC20(swapLpToken).approve(swap, stableTokensBalance);
+        uint256[] memory stableTokensToRemoveAmounts = ISwap(swap).calculateRemoveLiquidity(stableTokensBalance);
```

Comment:
you can remove this part, stableswap is different than Uniswapv2 on how it transfers fees
stableswap sends baseTokens 
uniswapV2 sends LP tokens 

so no need to swap LP tokens from the `StableTriMaker` contract

Reply:
oh oops, makes sense mb

Reply:
i.e. after calling `withdrawAdminFees` you will directly get the base tokens
eg here: https://github.com/trisolaris-labs/trisolaris_core/blob/main/test/stableswap/swap.test.ts#L1335